ai-platform/snippets/imagen/editImageInpaintingInsertMask.js

```diff
@@ -0,0 +1,120 @@
+  const {PredictionServiceClient} = aiplatform.v1;
+
+  // Import the helper module for converting arbitrary protobuf.Value objects
+  const {helpers} = aiplatform;
+
+  // Specifies the location of the api endpoint
+  const clientOptions = {
+    apiEndpoint: `${location}-aiplatform.googleapis.com`,
+  };
+
+  // Instantiates a client
+  const predictionServiceClient = new PredictionServiceClient(clientOptions);
+
+  const fs = require('fs');
+  const util = require('util');
+
+  async function editImageInpaintingInsertMask() {
+    // Configure the parent resource
+    const endpoint = `projects/${projectId}/locations/${location}/publishers/google/models/imagegeneration@006`;
+
+    const imageFile = fs.readFileSync(inputFile);
+    // Convert the image data to a Buffer and base64 encode it.
+    const encodedImage = Buffer.from(imageFile).toString('base64');
+
+    const maskImageFile = fs.readFileSync(maskFile);
+    // Convert the image mask data to a Buffer and base64 encode it.
+    const encodedMask = Buffer.from(maskImageFile).toString('base64');
+
+    const promptObj = {
+      prompt: prompt, // The text prompt describing what you want to see inserted
+      editMode: 'inpainting-insert',
+      image: {
+        bytesBase64Encoded: encodedImage,
+      },
+      mask: {
+        image: {
+          bytesBase64Encoded: encodedMask,
+        },
+      },
+    };
+    const instanceValue = helpers.toValue(promptObj);
+    const instances = [instanceValue];
+
+    const parameter = {
+      // Optional parameters
+      seed: 100,
+      // Controls the strength of the prompt
+      // 0-9 (low strength), 10-20 (medium strength), 21+ (high strength)
+      guidanceScale: 21,
+      sampleCount: 1,
+    };
+    const parameters = helpers.toValue(parameter);
+
+    const request = {
+      endpoint,
+      instances,
+      parameters,
+    };
+
+    // Predict request
+    const [response] = await predictionServiceClient.predict(request);
+    const predictions = response.predictions;
+    if (predictions.length === 0) {
+      console.log(
+        'No image was generated. Check the request parameters and prompt.'
+      );
+    } else {
+      let i = 1;
+      for (const prediction of predictions) {
+        const buff = Buffer.from(
+          prediction.structValue.fields.bytesBase64Encoded.stringValue,
+          'base64'
+        );
+        // Write image content to the output file
+        const writeFile = util.promisify(fs.writeFile);
+        const filename = `output${i}.png`;
+        await writeFile(filename, buff);
+        console.log(`Saved image ${filename}`);
+        i++;
+      }
+    }
+  }
+  await editImageInpaintingInsertMask().catch(err => {
+    console.error(err.message);
+    process.exitCode = 1;
+  });
+  // [END generativeaionvertexai_imagen_edit_image_inpainting_insert_mask]
+};
+
+// node editImageInpaintingInsertMask.js <inputFile> <maskFile> <prompt>
+main(...process.argv.slice(2));
```

Comment:
Our preference for samples is, if possible, to avoid burying details of what arguments are being supplied inside of a test, particularly since the region tags don't include the outer function definition with its parameters. Also, to avoid warnings in an IDE for an unhandled promise, it's best to at least catch errors. The following is preferred:

```javascript
main().catch(err => {
  console.error(err);
  process.exitcode = 1;
}
```

Please update all your samples accordingly with this guidance. Thanks!

Reply:
Refactored